"Multipart Content Without Name Error" When Adding Swagger Routes

**Describe the bug**
When I run the following code:

```scala
import zio.http.codec.QueryCodec
import zio.http.endpoint.Endpoint
import zio.http.endpoint.openapi.{OpenAPIGen, SwaggerUI}
import zio.http.{RoutePattern, Routes, Server, handler}
import zio.{Scope, ZIOAppDefault}

object EndpointExample extends ZIOAppDefault {
  val endpoint =
    Endpoint(RoutePattern.POST)
      .in[Int]("foo")
      .in[Boolean]("bar")
      .query(QueryCodec.query("q"))
      .out[Unit]

  val swaggerRoutes = SwaggerUI.routes("docs/openapi", OpenAPIGen.fromEndpoints(endpoint))
  val mainAppRoutes = Routes(endpoint.implement(handler((in1: Int, in2: Boolean, query: String) => ())))

  def run = Server
    .serve((mainAppRoutes ++ swaggerRoutes).toHttpApp)
    .provide(Server.default, Scope.default)
}
```

The following error will occur:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at example.EndpointExample.main(EndpointExample.scala)
Caused by: java.lang.IllegalArgumentException: Multipart content without name.
	at zio.http.endpoint.openapi.OpenAPIGen$.contentAsJsonSchema(OpenAPIGen.scala:340)
	at zio.http.endpoint.openapi.OpenAPIGen$.$anonfun$schemaByStatusAndMediaType$1(OpenAPIGen.scala:779)
	at zio.Chunk$Arr.mapChunk(Chunk.scala:1753)
	at zio.ChunkLike.map(ChunkLike.scala:121)
	at zio.ChunkLike.map$(ChunkLike.scala:120)
	at zio.Chunk.map(Chunk.scala:42)
	at zio.http.endpoint.openapi.OpenAPIGen$.schemaByStatusAndMediaType(OpenAPIGen.scala:775)
	at zio.http.endpoint.openapi.OpenAPIGen$.gen(OpenAPIGen.scala:485)
	at zio.http.endpoint.openapi.OpenAPIGen$.$anonfun$fromEndpoints$3(OpenAPIGen.scala:459)
	at scala.collection.immutable.List.map(List.scala:246)
	at scala.collection.immutable.List.map(List.scala:79)
	at zio.http.endpoint.openapi.OpenAPIGen$.fromEndpoints(OpenAPIGen.scala:459)
	at zio.http.endpoint.openapi.OpenAPIGen$.fromEndpoints(OpenAPIGen.scala:435)
	at example.EndpointExample$.<clinit>(EndpointExample.scala:17)
	... 1 more
```

**Expected behavior**
I expect the server to start properly without any errors.

**Additional context**
When I change the order endpoint properties, the problem will be fixed:

```scala
val endpoint =
  Endpoint(RoutePattern.POST)
    .query(QueryCodec.query("q"))
    .in[Int]("foo")
    .in[Boolean]("bar")
    .out[Unit]
```

--- a/zio-http/jvm/src/test/scala/zio/http/endpoint/openapi/OpenAPIGenSpec.scala
+++ b/zio-http/jvm/src/test/scala/zio/http/endpoint/openapi/OpenAPIGenSpec.scala
@@ -2442,6 +2442,18 @@ object OpenAPIGenSpec extends ZIOSpecDefault {
                                        |    "components" : {}
                                        |}""".stripMargin))
       },
+      test("Non content codecs are ignored when building multipart schema") {
+        // We only test there is no exception when building the schema
+        val endpoint =
+          Endpoint(RoutePattern.POST / "post")
+            .in[Int]("foo")
+            .in[Boolean]("bar")
+            .query(QueryCodec.query("q"))
+            .out[Unit]
+
+        SwaggerUI.routes("docs/openapi", OpenAPIGen.fromEndpoints(endpoint))
+        assertCompletes
+      },
     )
 
 }

--- a/zio-http/shared/src/main/scala/zio/http/endpoint/openapi/OpenAPIGen.scala
+++ b/zio-http/shared/src/main/scala/zio/http/endpoint/openapi/OpenAPIGen.scala
@@ -340,6 +340,20 @@ object OpenAPIGen {
                   .nullable(optional(metadata))
                   .description(description(metadata))
                   .annotate(annotations)
+              case (JsonSchema.Object(p, _, r), JsonSchema.Null)                =>
+                JsonSchema
+                  .Object(p, Left(false), r)
+                  .deprecated(deprecated(metadata))
+                  .nullable(optional(metadata))
+                  .description(description(metadata))
+                  .annotate(annotations)
+              case (JsonSchema.Null, JsonSchema.Object(p, _, r))                =>
+                JsonSchema
+                  .Object(p, Left(false), r)
+                  .deprecated(deprecated(metadata))
+                  .nullable(optional(metadata))
+                  .description(description(metadata))
+                  .annotate(annotations)
               case _ => throw new IllegalArgumentException("Multipart content without name.")
             }
 
@@ -798,8 +812,7 @@ object OpenAPIGen {
         (
           statusOrDefault,
           (
-            AtomizedMetaCodecs
-              .flatten(codec),
+            AtomizedMetaCodecs.flatten(codec),
             contentAsJsonSchema(codec, referenceType = referenceType) _,
           ),
         )

